src/client/client.rs

```diff
@@ -242,7 +242,7 @@ where
             if req.version() == Version::HTTP_2 {
                 warn!("Connection is HTTP/1, but request requires HTTP/2");
                 return Err(ClientError::Normal(
-                    crate::Error::new_user_unsupported_version(),
+                    crate::Error::new_user_unsupported_version().with_client_connect_info(pooled.conn_info.clone()),
```

Comment:
What specific info would you want to collect? Would it be feasible if we included some of that info in the `warn!`, above? Or perhaps right next to it, as a `trace!`?

Reply:
No, we have our own logging directives from the errors we get and create, we never use the `warn`/`trace` stuff from Hyper itself because we don't control it. We need to be able to access that data from the error itself.

The specific info we need comes from the `Connected` value of the streams returned by our own connector, so logging in Hyper itself wouldn't help anywya.

Reply:
> ... we never use the warn/trace stuff from Hyper itself because we don't control it.

I'm wondering if the current proposal to add a tracing instrumentation abstraction layer would be enough to bring Hyper spans and traces within your control.
To elaborate, the proposal is to add something like this (`client_send` is vapour-ware):

```rust
#[client_send(level=tracing::Level::Trace, name="Client::send_request", skip=[self,req])]
async fn send_request(
```

Hyper might default to using some proc-macro crate that covers the 80-20 rule. I wonder if, longer term, it wouldn't be possible for users to develop their own span/trace handling crates.

Admittedly, this would mean developing a bespoke proc-macro crate for your needs.  
Is this nuts or could it be useful?

Reply:
I don't think this would be useful. As stated before, even if we had access to the spans and traces, we also need access to the extra data stored in the connection's `Connected` value.

Reply:
> .... even if we had access to the spans and traces, we also need access to the extra data stored in the connection's `Connected` value

Understood, I believe you would.  

In your `client_send` you'd inspect for `send_request`, and then do as your PR here does replace:
```rust
crate::Error::new_user_unsupported_version(),
```

with
```rust                 crate::Error::new_user_unsupported_version().with_client_connect_info(pooled.conn_info.clone())
```

It might be possible for Hyper to aid this type of use with some place holder to search for , etc.  But even as it stands now this should be possible?

Reply:
I'm sorry I don't understand what you are getting at, but I'll note we are not "waiting for a PR to get accepted", we are using this in production already.